src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceMarginHelpers.cs

```diff
@@ -51,5 +56,74 @@ public static ImageMoniker GetMoniker(InheritanceRelationship inheritanceRelatio
             // The relationship is None. Don't know what image should be shown, throws
             throw ExceptionUtilities.UnexpectedValue(inheritanceRelationship);
         }
+
+        public static ImmutableArray<InheritanceMenuItemViewModel> CreateMenuItemViewModelsForSingleMember(ImmutableArray<InheritanceTargetItem> targets)
+        {
+            var targetsByRelationship = targets.OrderBy(target => target.DisplayName).GroupBy(target => target.RelationToMember).ToImmutableArray();
+            if (targetsByRelationship.Length == 1)
+            {
+                // If all targets have one relationship.
+                // e.g. interface IBar { void Bar(); }
+                // class A : IBar { void Bar() {} }
+                // class B : IBar { void Bar() {} }
+                // for 'IBar', the margin would be I↓. So header is not needed.
+                return targetsByRelationship[0].SelectAsArray(target => TargetMenuItemViewModel.Create(target, indent: false)).CastArray<InheritanceMenuItemViewModel>();
+            }
+            else
+            {
+                // Otherwise, it means these targets has different relationship,
+                // these targets would be shown in group, and a header should be shown as the first item to indicate the relationship to user.
+                using var _ = CodeAnalysis.PooledObjects.ArrayBuilder<InheritanceMenuItemViewModel>.GetInstance(out var builder);
+                foreach (var (relationship, targetItems) in targetsByRelationship)
+                {
+                    builder.AddRange(CreateMenuItemsWithHeader(relationship, targetItems));
+                }
+
+                return builder.ToImmutable();
+            }
+        }
+
+        public static ImmutableArray<InheritanceMenuItemViewModel> CreateMenuItemViewModelsForMultipleMembers(ImmutableArray<InheritanceMarginItem> members)
```

Comment:
Splitting this and `CreateMenuItemViewModelsForSingleMember` seems odd. They have the same input and output. Can they be merged? If not, an xml comment on the method would be helpful for knowing when they differ. The `Contract.ThrowIfTrue` below doesn't quite explain it.

Reply:
Yup those two methods could be merged. I will do that in next commit

Reply:
I'll edit the comment here, also I feel this might be a factory method for a class, not a helper ; )